spec/helpers/finance_helper_spec.rb

```diff
@@ -32,4 +32,28 @@
       end
     end
   end
+
+  describe "#cutoff_date" do
+    FinanceHelper::MILESTONE_DATES.each_with_index do |_date, index|
+      next if index == 0
+
+      milestone_begin_date = Date.parse(FinanceHelper::MILESTONE_DATES[index - 1])
+      milestone_end_date = Date.parse(FinanceHelper::MILESTONE_DATES[index]) - 1.day
+      random_date = rand(milestone_begin_date..milestone_end_date)
+
+      [milestone_begin_date, milestone_end_date, random_date].each do |date|
+        it "returns correct milestone cutoff date" do
+          travel_to(date) do
+            expect(cutoff_date).to eq(Date.parse(FinanceHelper::MILESTONE_DATES[index]).strftime("%-d %B %Y"))
+          end
+        end
+
+        it "returns correct payment period" do
+          travel_to(date) do
+            expect(payment_period).to eq(FinanceHelper::MILESTONE_DATES[index - 1, 2])
+          end
+        end
+      end
```

Comment:
Can we add the date in test description? And perhaps we want to add the index too?

`"milestone date #{index} on the day #{date} returns correct ..."`?